test/integration/smart-loan.test.ts

```diff
@@ -424,5 +430,170 @@ describe('Smart loan', () => {
+      expect(fromWei(await wrappedLoan.getDebt())).to.be.closeTo(300, 0.1);
+      expect(await wrappedLoan.getSolvencyRatio()).to.be.closeTo(BigNumber.from(1333), 1);
+    });
+
+    it("should invest", async () => {
+
+
+      await wrappedLoan.invest(toBytes32('USD'), toWei("9000", usdTokenDecimalPlaces));
+      await wrappedLoan.invest(toBytes32('LINK'), toWei("10000", usdTokenDecimalPlaces));
+
+      let balances = await wrappedLoan.getAllAssetsBalances();
+
+      const currentUSDTokenBalance = balances[0];
+      const currentLINKTokenBalance = balances[1];
+      expect(currentUSDTokenBalance).to.be.equal(toWei("9000", usdTokenDecimalPlaces));
+      expect(currentLINKTokenBalance).to.be.equal(toWei("10000", usdTokenDecimalPlaces));
+    });
+
+    it("should fail a sellout attempt", async () => {
+
+      let UPDATED_MOCK_PRICES = MOCK_PRICES.map(
+        (token: any) => {
+          if (token.symbol == 'USD' || token.symbol == 'LINK') {
+            token.value = 0.6 * token.value;
+          }
+          return token;
+        }
+      );
+
+      let loanUpdated = WrapperBuilder
+        .mockLite(loan)
+        .using(
+          () => {
+            return {
+              prices: UPDATED_MOCK_PRICES,
+              timestamp: Date.now()
+            }
+          })
+
+      let loan_solvent = await loanUpdated.isSolvent();
+      let loan_below_sellout_treshold = await loanUpdated.isBelowSelloutTreshold();
+      expect(loan_solvent).to.be.false;
+      expect(loan_below_sellout_treshold).to.be.false;
+      await expect(loanUpdated.sellout()).to.be.revertedWith('Cannot sellout a solvent account');
+    });
+
+    it("should sellout a loan covering whole debt", async () => {
+      let UPDATED_MOCK_PRICES = MOCK_PRICES.map(
+        (token: any) => {
+          if (token.symbol == 'USD' || token.symbol == 'LINK') {
+            token.value = 0.7 * token.value;
+          }
+          return token;
+        }
+      );
+
+      let loanUpdated = WrapperBuilder
+        .mockLite(loan)
+        .using(
+          () => {
+            return {
+              prices: UPDATED_MOCK_PRICES,
+              timestamp: Date.now()
+            }
+          })
+      expect(await loanUpdated.isSolvent()).to.be.false;
+      expect(await loanUpdated.isBelowSelloutTreshold()).to.be.true;
+
+      let balances = await loanUpdated.getAllAssetsBalances();
+      expect(balances[0]).to.be.equal(9000000000);
+      expect(balances[1]).to.be.equal(10000000000);
+
+      await loanUpdated.sellout();
+
+      expect(await loanUpdated.isSolvent()).to.be.true;
+      expect(await loanUpdated.isBelowSelloutTreshold()).to.be.false;
+
+      balances = await loanUpdated.getAllAssetsBalances();
+      expect(balances[0]).to.be.equal(0);
+      expect(balances[1]).to.be.equal(0);
+      expect(await loanUpdated.getDebt()).to.be.equal(0);
+      expect((await loanUpdated.getTotalValue()).gt(0)).to.be.true;
```

Comment:
Maybe it would be worth checking if value of pool increased? Meaning that repay transaction went well